### PR TITLE
Update rxfire and rxjs

### PR DESCRIPTION
**Pull request by jhuleatt**

### CODE REVIEW

**Comment by jhuleatt** [comment]
@jamesdaniels TS is angry, and I think it's due to the `add` hack you mentioned the other day. Can you please take a look?

Screen Shot 2021-05-14 at 12 58 45 PM

**Comment by jamesdaniels** [comment]
@jhuleatt you can't chain off `add` any-longer.

This should work:
```ts
this._innerSubscriber.add(this._reset);
return this._innerSubscriber;
```

**Comment by jhuleatt** [comment]
> @jhuleatt you can't chain off `add` any-longer.
> 
> This should work:

Thanks! That did. Just investigating a failing test and then this should be ready for review

**Review by jamesdaniels — APPROVED** [review]

File: package.json
Comment (jamesdaniels) [inline package.json:97]: We should probably keep 6 while feasible, since 6 => 7 is a big jump for developers.

E.g, `^6.6.3 || ^7.0.1`

We could add to the matrix test like we did RxFire. WDYT?

File: src/sdk.tsx
Comment (jhuleatt) [inline src/sdk.tsx:117]: because [`toPromise` is deprecated](https://rxjs.dev/deprecations/to-promise)

File: src/SuspenseSubject.ts
Comment (jhuleatt) [inline src/SuspenseSubject.ts:87]: @jamesdaniels this was causing useSDK to throw out its value after it loaded. Would it be bad if I removed it?

Comment (jhuleatt) [reply src/SuspenseSubject.ts:87]: This could be important for other functions like `useUser`

Suggestion from James: have a flag `resetOnUnsubscribe` that's true for everything except the `useSDK` functions

File: package.json
Comment (jhuleatt) [reply package.json:97]: Convo with James: keep 6 in for now to avoid version conflicts if someone is using 6 in their app

File: src/sdk.tsx
Comment (jhuleatt) [reply src/sdk.tsx:117]: note: this doesn't exist in rxjs 6

Comment (jhuleatt) [reply src/sdk.tsx:117]: TODO: when we have a major that drops rxjs 6 we can switch, for now we can just use `toPromise`

File: package.json
Comment (jhuleatt) [reply package.json]: done

File: src/sdk.tsx
Comment (jhuleatt) [reply src/sdk.tsx]: done

File: src/SuspenseSubject.ts
Comment (jhuleatt) [reply src/SuspenseSubject.ts:87]: added a test for this, and it passes without adding `reset`, so I'm removing that line

**Comment by jhuleatt** [comment]
I think this is blocked by https://github.com/FirebaseExtended/rxfire/issues/13

I can't get ReactFire's example web app working because of that issue, and I'd really like to click around in the example for peace of mind before merging

**Comment by jhuleatt** [comment]
> I think this is blocked by [FirebaseExtended/rxfire#13](https://github.com/FirebaseExtended/rxfire/issues/13)
> 
> I can't get ReactFire's example web app working because of that issue, and I'd really like to click around in the example for peace of mind before merging

I've fixed the rxfire issue and released `rxfire@5.0.0-rc.1`. With that build, now all the tests pass AND I'm able to click around in the example app! Everything looks good.

**Pull request merged**
